tests/main_test.py

```diff
@@ -4,47 +4,92 @@
+    },
+    {
+        'func': main.check_ping,
+        'expected_res': False,
+        'arg': ('wrongurlpage.com',)
+    }, 
+    {
+        'func': main.obtain_hostname,
+        'expected_res': 'example.com',
+        'arg': ('http://example.com',)
+    },
     {
         'func': main.get_users,
         'expected_res': ['12345678'],
         'arg': tuple(),
         'mock_obj': [main],
         'mock_func': ['getenv'],
-        'mock_ret': ['12345678'],
-        'is_async': False
+        'mock_ret': ['12345678']
     },
     {
         'func': main.get_users,
         'expected_res': ['12345678', '23456789'],
         'arg': tuple(),
         'mock_obj': [main],
         'mock_func': ['getenv'],
-        'mock_ret': ['12345678;23456789'],
-        'is_async': False
+        'mock_ret': ['12345678;23456789']
     },
     {
-        'func': main.check_ok,
+        'func': main.check_ping,
         'expected_res': True,
-        'arg': ('http://example.org/',),
-        'is_async': False
+        'arg': ('example.org',),
+        'mock_obj': [main.subprocess],
+        'mock_func': ['call'],
+        'mock_ret': [0]
     },
     {
-        'func': main.check_ok,
-        'expected_res': False,
-        'arg': ('https://github.com/404',),
-        'is_async': False
+        'func': main.make_request_to_telegram,
+        'expected_res': {"ok":False,"error_code":401,"description":"Unauthorized"},
+        'arg': ('http://example.com', 'get', '0'),
+        'mock_obj': [main],
+        'mock_func': ['getenv'],
+        'mock_ret': ['123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11'],
+        'is_async': True  
     },
     {
-        'func': main.check_ping,
-        'expected_res': True,
-        'arg': ('example.org',),
-        'is_async': False
+        'func': main.handle_communication,
+        'expected_res': None,
+        'arg': ('http://example.com', 'get'),
+        'mock_obj': [main, main, main],
+        'mock_func': ['get_users', 'run', 'sleep'],
+        'mock_ret': [['12345678'], {"ok":False,"error_code":429,"description":"Unauthorized"}, None],
     },
     {
-        'func': main.check_ping,
-        'expected_res': False,
-        'arg': ('wrongurlpage.com',),
-        'is_async': False
+        'func': main.handle_communication,
+        'expected_res': None,
+        'arg': ('http://example.com', 'get'),
+        'mock_obj': [main, main],
+        'mock_func': ['get_users', 'run'],
+        'mock_ret': [['12345678'], {"ok":True,"error_code":429,"description":"Unauthorized"}],
+    },
+    {
+        'func': main.handle_communication,
+        'expected_res': None,
+        'arg': ('http://example.com', 'get'),
+        'mock_obj': [main, main],
+        'mock_func': ['get_users', 'run'],
+        'mock_ret': [[], None],
+    },
+    {
+        'func': main.main,
+        'expected_res': None,
+        'arg': tuple(),
+        'mock_obj': [main],
+        'mock_func': ['handle_urls'],
+        'mock_ret': [None],
```

Comment:
Unneeded comma at the end of the last four test entries.

Reply:
Thanks, I didn't notice